pkg/reporter/csv.go

```diff
@@ -0,0 +1,130 @@
+package reporter
+
+import (
+	"encoding/csv"
+	"os"
+
+	"github.com/safedep/vet/pkg/analyzer"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/policy"
+)
+
+type CsvReportingConfig struct {
+	Path string
+}
+
+type csvReporter struct {
+	config          CsvReportingConfig
+	csvRecords      []csvRecord
+	summaryReporter Reporter
+	violations      map[string]*analyzer.AnalyzerEvent
+}
+
+type csvRecord struct {
+	ecosystem       string
+	manifestPath    string
+	packageName     string
+	packageVersion  string
+	violationReason string
+}
+
+func NewCsvReporter(config CsvReportingConfig) (Reporter, error) {
+	summaryReporter, _ := NewSummaryReporter()
+
+	return &csvReporter{
+		config:          config,
+		summaryReporter: summaryReporter,
+		csvRecords:      make([]csvRecord, 0),
+		violations:      make(map[string]*analyzer.AnalyzerEvent),
+	}, nil
+}
+
+func (r *csvReporter) Name() string {
+	return "CSV Report Generator"
+}
+
+func (r *csvReporter) AddManifest(manifest *models.PackageManifest) {
+	r.summaryReporter.AddManifest(manifest)
```

Comment:
Do we really need the summary reporter? I do not see any reference to `summaryReporter` while generating CSV. We can just ignore the manifest here I think.

Reply:
@abhisek Removed it.